gcc/rust/hir/tree/rust-hir-expr.h

```diff
@@ -24,6 +24,7 @@
 #include "rust-hir.h"
 #include "rust-hir-path.h"
 #include "rust-operators.h"
+#include <memory>
```

Comment:
I have a funny feeling this should be included by rust-system.h I think @tschwinge found I was including it and it was marked as poisoned or something in a bootstrap build.

Reply:
Yes, instead please (early) `#define INCLUDE_MEMORY`, see <https://github.com/Rust-GCC/gccrs/blob/2a274c1b7bfaa860f41b4bdcfb7008df0bc4cc26/gcc/system.h#L742-L748>.

Reply:
Hmmm, clangd did some nasty stuff behind my back. It automatically adds include files that break the build. I did not add any header file intentionally, so this one is probably the work of clangd.